frame/contracts/src/weights.rs

```diff
@@ -463,89 +466,105 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
+	// Storage: Skipped Metadata (r:0 w:0)
+	fn seal_set_storage_per_new_kb(n: u32, ) -> Weight {
+		(761_439_000 as Weight)
+			// Standard Error: 317_000
+			.saturating_add((32_500_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(T::DbWeight::get().reads(105 as Weight))
+			.saturating_add(T::DbWeight::get().writes(103 as Weight))
+	}
+	// Storage: Skipped Metadata (r:0 w:0)
+	fn seal_set_storage_per_old_kb(n: u32, ) -> Weight {
+		(775_774_000 as Weight)
+			// Standard Error: 374_000
+			.saturating_add((12_308_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(T::DbWeight::get().reads(105 as Weight))
+			.saturating_add(T::DbWeight::get().writes(103 as Weight))
 	}
 	// Storage: Skipped Metadata (r:0 w:0)
 	fn seal_clear_storage(r: u32, ) -> Weight {
-		(163_262_000 as Weight)
-			// Standard Error: 1_492_000
-			.saturating_add((799_087_000 as Weight).saturating_mul(r as Weight))
+		(99_200_000 as Weight)
+			// Standard Error: 961_000
+			.saturating_add((503_951_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 			.saturating_add(T::DbWeight::get().reads((100 as Weight).saturating_mul(r as Weight)))
 			.saturating_add(T::DbWeight::get().writes(3 as Weight))
 			.saturating_add(T::DbWeight::get().writes((100 as Weight).saturating_mul(r as Weight)))
 	}
 	// Storage: Skipped Metadata (r:0 w:0)
+	fn seal_clear_storage_per_kb(n: u32, ) -> Weight {
+		(742_815_000 as Weight)
+			// Standard Error: 259_000
+			.saturating_add((12_010_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(T::DbWeight::get().reads(105 as Weight))
+			.saturating_add(T::DbWeight::get().writes(103 as Weight))
+	}
+	// Storage: Skipped Metadata (r:0 w:0)
 	fn seal_get_storage(r: u32, ) -> Weight {
-		(288_728_000 as Weight)
-			// Standard Error: 752_000
-			.saturating_add((434_780_000 as Weight).saturating_mul(r as Weight))
+		(108_304_000 as Weight)
+			// Standard Error: 844_000
+			.saturating_add((444_124_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(4 as Weight))
 			.saturating_add(T::DbWeight::get().reads((100 as Weight).saturating_mul(r as Weight)))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	// Storage: Skipped Metadata (r:0 w:0)
+	fn seal_get_storage_per_kb(n: u32, ) -> Weight {
+		(695_381_000 as Weight)
+			// Standard Error: 376_000
+			.saturating_add((69_455_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(T::DbWeight::get().reads(104 as Weight))
+			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+	}
+	// Storage: Skipped Metadata (r:0 w:0)
 	fn seal_contains_storage(r: u32, ) -> Weight {
-		(209_606_000 as Weight)
-			// Standard Error: 1_137_000
-			.saturating_add((680_187_000 as Weight).saturating_mul(r as Weight))
+		(0 as Weight)
+			// Standard Error: 1_580_000
+			.saturating_add((736_285_000 as Weight).saturating_mul(r as Weight))
```

Comment:
Can you explain why this is higher than `seal_get_storage`? It looks like the weight will surpass `seal_get_storage`'s as `r` increases. 

The `*per_kb` weights make more sense (that is, `contains_storage` is always less than `get_storage`), so I wonder if I'm interpreting this benchmark wrong

Reply:
You are right this makes no sense. We only look at `r` in this case which makes `contains_storage` more expensive. I need to look into those benchmarks. Maybe there is a bug.